examples/luna/ntsb_loader.py

```diff
@@ -0,0 +1,148 @@
+    if "dateAndTime" not in doc.properties["entity"] and "dateTime" not in doc.properties["entity"]:
+        return doc
+    raw_date: str = doc.properties["entity"].get("dateAndTime") or doc.properties["entity"].get("dateTime")
+    raw_date = raw_date.replace("Local", "")
+    parsed_date = parser.parse(raw_date, fuzzy=True)
+    extracted_date = parsed_date.date()
+    doc.properties["entity"]["day"] = extracted_date.isoformat()
+    if parsed_date.utcoffset():
+        doc.properties["entity"]["isoDateTime"] = parsed_date.isoformat()
+    else:
+        doc.properties["entity"]["isoDateTime"] = parsed_date.isoformat() + "Z"
+
+    return doc
+
+
+if os.path.exists("/.dockerenv"):
+    opensearch_host = "opensearch"
+    print("Assuming we are in a Sycamore Jupyter container, using opensearch for OpenSearch host")
+else:
+    opensearch_host = "localhost"
+    print("Assuming we are running outside of a container, using localhost for OpenSearch host")
+
+os_client_args = {
+    "hosts": [{"host": opensearch_host, "port": 9200}],
+    "http_compress": True,
+    "http_auth": ("admin", "admin"),
+    "use_ssl": True,
+    "verify_certs": False,
+    "ssl_assert_hostname": False,
+    "ssl_show_warn": False,
+    "timeout": 120,
+}
+
+os_client = OpenSearch(**os_client_args)  # type: ignore
+if os_client.indices.exists(index=INDEX):
+    if args.delete:
+        print(f"Index {INDEX} already exists, deleting as requested")
+        os_client.indices.delete(index=INDEX)
+    else:
+        raise Exception(f"Index {INDEX} already exists. Run with --delete to delete it.")
+
+index_settings = {
+    "body": {
+        "settings": {"index.knn": True, "number_of_shards": 5, "number_of_replicas": 1},
+        "mappings": {
+            "properties": {
+                "embedding": {
+                    "dimension": 384,
+                    "method": {
+                        "engine": "faiss",
+                        "space_type": "l2",
+                        "name": "hnsw",
+                        "parameters": {},
+                    },
+                    "type": "knn_vector",
+                }
+            }
+        },
+    }
+}
+
+
+context = sycamore.init()
+docset = context.read.binary(SOURCE_DATA_PATH, binary_format="pdf")
+tokenizer = HuggingFaceTokenizer("thenlper/gte-small")
+llm = OpenAI(OpenAIModels.GPT_3_5_TURBO.value)
+
+# partitioning docset
+partitioned_docset = (
+    docset.partition(partitioner=UnstructuredPdfPartitioner())
+    # these are here mostly as examples; the last materialize will at this point take
+    # effect, you can use these during testing.
+    .materialize(path="/tmp/after_partition", source_mode=sycamore.MaterializeSourceMode.IF_PRESENT)
+    .map(add_property_to_schema)
+    .extract_properties(property_extractor=OpenAIPropertyExtractor(llm=llm, num_of_elements=35))
+    .materialize(path="/tmp/after_llm", source_mode=sycamore.MaterializeSourceMode.IF_PRESENT)
```

Comment:
Why are we doing this here - just as an example? Seems like this should be an option or the directories we're materializing to should be using `tempdir` (or at least named constants at the top of this file).

Reply:
They're there as examples and to help while I was getting it working.